### PR TITLE
Add new CLI option `debug-print-ast`

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Adds new CLI option `debug-print-ast` for debugging.

It is useful to use it  with `jq`.

like this:

```sh
./bin/prettier.js --debug-print-ast | jq '.program | walk(if type == "object" and has("loc") then del(.loc, .start, .end)  else . end)'
```

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/integration/__tests__/debug-print-ast.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+const runPrettier = require("../runPrettier.js");
+
+describe("prints information for debugging AST --debug-print-comments", () => {
```
Comment (thorn0) [inline tests/integration/__tests__/debug-print-ast.js:5]:
```suggestion
describe("prints information for debugging AST --debug-print-ast", () => {
```

File: tests/integration/__tests__/__snapshots__/debug-print-ast.js.snap [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
...\":11,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":10},\\"end\\":{\\"line\\":1,\\"column\\":11}}},{\\"type\\":{\\"label\\":\\"string\\",\\"beforeExpr\\":false,\\"startsExpr\\":true,\\"rightAssociative\\":false,\\"isLoop\\":false,\\"isAssign\\":false,\\"prefix\\":false,\\"postfix\\":false,\\"binop\\":null,\\"updateContext\\":null},\\"value\\":\\"foo\\",\\"start\\":12,\\"end\\":17,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":12},\\"end\\":{\\"line\\":1,\\"column\\":17}}},{\\"type\\":{\\"label\\":\\";\\",\\"beforeExpr\\":true,\\"startsExpr\\":false,\\"rightAssociative\\":false,\\"isLoop\\":false,\\"isAssign\\":false,\\"prefix\\":false,\\"postfix\\":false,\\"binop\\":null,\\"updateContext\\":null},\\"start\\":17,\\"end\\":18,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":17},\\"end\\":{\\"line\\":1,\\"column\\":18}}},{\\"type\\":{\\"label\\":\\"eof\\",\\"beforeExpr\\":false,\\"startsExpr\\":false,\\"rightAssociative\\":false,\\"isLoop\\":false,\\"isAssign\\":false,\\"prefix\\":false,\\"postfix\\":false,\\"binop\\":null,\\"updateContext\\":null},\\"start\\":18,\\"end\\":18,\\"loc\\":{\\"start\\":{\\"line\\":1,\\"column\\":18},\\"end\\":{\\"line\\":1,\\"column\\":18}}}]}}"`;
```
Comment (thorn0) [inline tests/integration/__tests__/__snapshots__/debug-print-ast.js.snap:3]: Is it intended that the output includes `text`, not only `ast`?

Comment (sosukesuzuki) [reply tests/integration/__tests__/__snapshots__/debug-print-ast.js.snap]: It is intended when I wrote this PR, but I thought about it and decided I didn't need it, so I deleted it fe3be6b

**Review by thorn0 — APPROVED** [review]

**Comment by jasikpark** [comment]
Is there a way to use this w/ the JS API? 
https://github.com/snowpackjs/prettier-plugin-astro/ - trying to inspect the docs I'm generating w/ less luck

**Pull request merged**